canonical_sphinx/config.py

```diff
@@ -77,9 +77,42 @@ def config_inited(_app: Sphinx, config: Any) -> None:  # noqa: ANN401
         ],
     )
 
+    # The URL prefix for the notfound extension depends on whether the documentation uses versions.
+    # For documentation on documentation.ubuntu.com, we also must add the slug.
+    url_version = ""
+    url_lang = ""
+
+    # Determine if the URL uses versions and language
+    if (
+        "READTHEDOCS_CANONICAL_URL" in os.environ
+        and os.environ["READTHEDOCS_CANONICAL_URL"]
+    ):
```

Comment:
yes I'd rather keep it as synced with upstream as possible, specially because in this case this should be a config option of the extension itself (so I'd like to change this code soon)